### ISSUE TITLE
and, or, & xor logical operators aren't checked against variable-setting

### ISSUE THREAD

**Issue by sybrew** [sirbrillig/phpcs-variable-analysis#99]
Code speaks a thousand words. Replace `and` with the other logical operators (`or`, `xor`), and the same error will yield.

```php
$get_something_and_do_stuff = true;

$get_something_and_do_stuff
	and $something = get_something()
	and $something === 'something' // VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
	and do_stuff();
```

```
Variable $something is undefined.
(VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
```

**Comment by sirbrillig** [comment]
Oh, interesting. I hadn't considered a single statement defining and using a variable. Good find.

**Comment by sybrew** [comment]
I don't think it's feasible to create a fix for this, however. Consider these scenarios, where every instance of true/false may be a function or variable call which you can't test.

Positives:

```php
false
	and $something = true
	or  'value' === $something; // warning
```

```php
true
	and $something = true
	or 'value' === $something; // warning, never runs.
```

```php
false
	or $something = false
	and 'value' === $something; // warning, never runs.
```

-------

False positives:

```php
false
	or $something = true
	and 'value' === $something; // warning that's incorrect
```

--------

False negatives:

```php
false
	and $something = true;

if ( $something ) { // no warning, but it's undefined...
	return false;
}
```
